### PR TITLE
Add prowjob/periodic-openshift-release-qci-pruner

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-4041

/cc @openshift/test-platform @jupierce 

/hold

we need to set up the token in Vault.

### CODE REVIEW

File: ci-operator/jobs/infra-periodics.yaml [hunk 1]
```diff
@@ -2690,6 +2690,29 @@ periodics:
...
           cpu: 500m
+- agent: kubernetes
+  cluster: app.ci
+  cron: '@daily'
+  decorate: true
+  name: periodic-openshift-release-qci-pruner
+  spec:
+    containers:
+    - args:
+      - --confirm=false
```
Comment (hongkailiu) [inline ci-operator/jobs/infra-periodics.yaml:2701]: We will make it to true when we are happy with the logs with `false`.

File: core-services/ci-secret-bootstrap/_config.yaml [hunk 2]
```diff
@@ -5474,6 +5474,14 @@ secret_configs:
   - cluster: app.ci
     name: qci-pull-through-cache-credentials
     namespace: ci
+- from:
+    token:
+      field: token
```
Comment (hongkailiu) [inline core-services/ci-secret-bootstrap/_config.yaml:5479]: We need to set up the Vault item accordingly.

File: clusters/app.ci/supplemental-ci-images/qci-pruner.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    successfulBuildsHistoryLimit: 1
+    failedBuildsHistoryLimit: 2
+    output:
+      to:
+        kind: ImageStreamTag
+        name: qci-pruner:latest
+    source:
+      dockerfile: |
+        FROM registry.access.redhat.com/ubi9/python-312:latest
+        RUN pip install python-dxf==11.4.0 && \
```
Comment (jupierce) [inline clusters/app.ci/supplemental-ci-images/qci-pruner.yaml:24]: python-dxf is no longer a dep

File: ci-operator/jobs/infra-periodics.yaml [hunk 4]
```diff
@@ -2690,6 +2690,29 @@ periodics:
...
+      command:
+      - ./qci_registry_pruner.py
+      env:
+      - name: QUAY_OAUTH_TOKEN_ENV_NAME
```
Comment (jupierce) [inline ci-operator/jobs/infra-periodics.yaml:2705]:
```suggestion
      - name: QUAY_OAUTH_TOKEN
```

File: clusters/app.ci/supplemental-ci-images/qci-pruner.yaml (hunk 3 above)
Comment (deepsm007) [reply clusters/app.ci/supplemental-ci-images/qci-pruner.yaml:24]: can we use `ubi9/ubi-minimal:latest`  ?

File: ci-operator/jobs/infra-periodics.yaml (hunk 4 above)
Comment (hongkailiu) [reply ci-operator/jobs/infra-periodics.yaml:2705]: Should we keep it the same as https://github.com/openshift/release/pull/53182/files#diff-5d7c76ce5aadd38b6f0b95f1f529c461c621319613ecc02ecd3c9aefd73181afR92 ?

File: clusters/app.ci/supplemental-ci-images/qci-pruner.yaml (hunk 3 above)
Comment (hongkailiu) [reply clusters/app.ci/supplemental-ci-images/qci-pruner.yaml:24]: > python-dxf is no longer a dep

Could do you a freeze and update https://github.com/jupierce/testplatform-qci-registry-pruner/blob/main/requirements.txt accordingly. 
I cannot really see the requirements here from the code.

Comment (hongkailiu) [reply clusters/app.ci/supplemental-ci-images/qci-pruner.yaml:24]: > can we use `ubi9/ubi-minimal:latest` ?

We need python for this this as @jupierce implements the pruner with it in https://github.com/openshift/release/pull/53182.

File: core-services/ci-secret-bootstrap/_config.yaml (hunk 2 above)
Comment (hongkailiu) [reply core-services/ci-secret-bootstrap/_config.yaml:5479]: Done.

File: clusters/app.ci/supplemental-ci-images/qci-pruner.yaml (hunk 3 above)
Comment (hongkailiu) [reply clusters/app.ci/supplemental-ci-images/qci-pruner.yaml]: Tried the `imports` in the base image.
It seems there is no dependencies at all.

File: ci-operator/jobs/infra-periodics.yaml (hunk 1 above)
Comment (jupierce) [reply ci-operator/jobs/infra-periodics.yaml:2701]: This argument parser does not support an explicit `--confirm=false`. `--confirm` just needs to be omitted.

Comment (hongkailiu) [reply ci-operator/jobs/infra-periodics.yaml]: Removed.

**Pull request merged**